### PR TITLE
added in back-end for handling username and/or email if they're alrea…

### PR DESCRIPTION
**Pull request by joseph-w-valdez**
image

### CODE REVIEW

**Comment by joseph-w-valdez** [comment]
image

**Comment by joseph-w-valdez** [comment]
image
image

**Comment by joseph-w-valdez** [comment]
image
image

**Comment by joseph-w-valdez** [comment]
It works currently with front-end integration, too. I think this is as much error handling I wanted for basic user registration.

File: client/pages/Register.jsx [hunk 1]
```diff
@@ -19,7 +19,18 @@ const Register = ({ onMount }) => {
       })
       .catch((error) => {
         console.error(error);
-        setErrorMessage('An error occurred while fetching data. Please try again later.');
+        if (error.response && error.response.status === 409) {
```
Comment (RobertGardner) [inline client/pages/Register.jsx:22]: Nice! I love to see good error checking and reporting.

File: server/index.js [hunk 2]
```diff
@@ -65,17 +65,45 @@ app.post('/marvel/registration', (req, res, next) => {
...
+        select "id", "username", "email", "createdAt" from "users" where "username" = $1 limit 1
+      `;
+      const usernameParams = [username];
+
+      const emailSql = `
+        select "id", "username", "email", "createdAt" from "users" where "email" = $1 limit 1
+      `;
+      const emailParams = [email];
+
+      return Promise.all([
```
Comment (RobertGardner) [inline server/index.js:78]: Running in parallel. Nice!

File: server/index.js [hunk 3]
```diff
@@ -65,17 +65,45 @@ app.post('/marvel/registration', (req, res, next) => {
...
     })
     .then((result) => {
-      const [user] = result.rows;
-      res.status(201).json(user);
+      if (result.rowCount === 0) {
+        throw new ClientError(409, 'username or email already exists');
+      } else {
+        const [user] = result.rows;
+        res.status(201).json(user);
+      }
```
Comment (RobertGardner) [inline server/index.js:106]: All this code would look much simpler with `async/await`. You should make that transition after you've got your features finished.

**Review by RobertGardner — APPROVED** [review]

File: server/index.js (hunk 3 above)
Comment (joseph-w-valdez) [reply server/index.js:106]: Will do!

**Pull request merged**